### ISSUE TITLE
Improve karma test detection

### ISSUE THREAD

**Issue by Kjir** [janko-m/vim-test#127]
Right now, the karma runner matches the same files as jasmine. The implementation should be improved to actually check for the correct binary. As suggested by @janko-m the [TAP runner](https://github.com/janko-m/vim-test/blob/ee4099c252a27da232815544da2832ae1d9a8580/autoload/test/javascript/tap.vim#L14) already lays the groundwork for this.

**Comment by janko-m** [comment]
@Kjir Definitely, would you mind making a PR for that?

**Comment by Kjir** [comment]
Work is in progress! 😄

**Comment by Kjir** [comment]
Now I am much more satisfied with how things work! I moved Karma before Jasmine and Mocha since it is a runner for both of those: now that the detection for karma is smarter, it is able to tell whether we want to run tests with karma or with mocha/jasmine. To do that it needs to have higher priority.

I also wanted to introduce a variable to select the karma reporters, but then I discovered that vim-test already has the possibility to define custom parameters with `g:test#javascript#karma#options`, so I ditched that part of code! Awesome! 😄 

I had to introduce `g:test#javascript#karma#conffile` with a default of `karma.conf.js` to define the name of that karma config file. I could have left it up to the user to define it in `g:test#javascript#karma#options`, but then I wouldn't be able to check for its existence. Hence the new variable.
Maybe there are smarter ways to do it without a new configuration variable, but none came to mind.
